### PR TITLE
Require explicit registration; publish as ESM format

### PR DESCRIPTION
**Pull request by keller-mark**
Change list
- Require explicit registration of the datafetcher by the consumer app/package; do not use `higlass-register` implicitly in `index.js` / via side effect
- Use esbuild for bundling
- Use Vite for development server + simultaneous esbuild in watch-mode (to deal with `Buffer` polyfilling)
- Use Vitest for testing
- Bump version of package
- Bump version of NodeJS in GitHub Actions workflows

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -5,27 +5,26 @@ name: Node.js CI
...
   build:
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
-        node-version: [12.x]
+        node-version: [18, 20]
 
     steps:
       - uses: actions/checkout@v2
```
Comment (alexander-veit) [inline .github/workflows/nodejs.yml:21]: Do these actions need a new version as well (v3)? I thought that is needed when using a higher node version

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "higlass-bigwig-datafetcher",
-  "version": "0.1.2",
+  "version": "0.1.3",
```
Comment (alexander-veit) [inline package.json]: I think a minor version bump is appropriate.

**Review by alexander-veit — APPROVED** [review]
Thanks a lot for this update! I seems to work well.

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (alexander-veit) [reply .github/workflows/nodejs.yml:21]: Well, it's building. So it's probably not required.

Comment (keller-mark) [reply .github/workflows/nodejs.yml:21]: Good point, I have updated the action version just in case

**Comment by keller-mark** [comment]
Thanks for the review! I do not have permission to merge - would you be able to do that if everything looks alright to you?

**Pull request merged**